packages/mobile/src/transactions/TransactionsList.tsx

```diff
@@ -234,7 +239,22 @@ export class TransactionsList extends React.PureComponent<Props> {
     if (transactions.length < 1) {
       return
     }
-
+    // Transaction list has changed and we need to update the balance.
+    this.props.fetchDollarBalance()
+
+    // For good UX we want new transactions appear at the same time we update
+    // balance. To achieve this we optimistically re-calculate balance,
+    // based on the new transactions and show it to a user until we have
+    // new balance from the contract
+    if (this.props.dollarBalanceLastFetch) {
+      const newBalance = transactions.reduce((acc, tx) => {
+        if (tx.timestamp * 1000 > this.props.dollarBalanceLastFetch!) {
+          return acc.plus(tx.amount.value)
+        }
+        return acc
+      }, new BigNumber(this.props.dollarBalance || 0))
+      this.props.setBalance(newBalance.toString())
+    }
```

Comment:
`TransactionsList` is used for both the cUSD and cGLD feeds (home and gold tabs).
`fetchDollarBalance`/`dollarBalanceLastFetch`/`setBalance` should be made generic and dependant on the `currency` prop passed to `TransactionsList`, as is it will set an incorrect cUSD balance when the component is used from the gold tab.

I see the call here to refresh the balance is going to duplicate the work done in https://github.com/celo-org/celo-monorepo/blob/25feb786595f56c1f88c9ec99be0f3f5dcbf91c6/packages/mobile/src/home/saga.ts#L68 since the GraphQL query is also doing polling. Looks like we could skip the manual fetch here given this info.

I think there's an edge case leading to an incorrect balance if this callback is called more than once and the `fetchDollarBalance` from the contract hasn't had time to run.
Can we make the optimistic balance update be resistant to this?

Reply:
Ah my bad, I see `setBalance` also updates `lastFetch` so please ignore my comment about the edge case.

Reply:
`tx.timestamp` is already in milliseconds, we shouldn't multiply it.

Reply:
Ah, ok. I've just looked in tests: https://github.com/celo-org/celo-monorepo/blob/cd7be93a3abd1333d7972765f23cfa70d17cc492/packages/mobile/src/transactions/TransactionsList.test.tsx#L71. Should we update the mocks then?